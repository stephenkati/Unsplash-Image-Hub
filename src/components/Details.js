import { useSelector } from "react-redux";

const Details = () => {
  const details = useSelector((state) => state.collectionDetails.collectionDetails);

  return (
    <div style={{color:"white"}} key={details.title} className="details-page">
      <div className="details-header">
        <img src={details.cover_photo.urls.regular} />
        <h2>{details.title}</h2> 
      </div>
      <div className="collection-stats">
        <h3>Collection Stat's</h3>
      </div>
      <ul className="details-list">
        <li className="details-list-item">
          <span>Title</span>
          <span>{details.title}</span>
        </li>
        {!details.description && (
          <li className="details-list-item">
            <span>Description</span>
            <span>null</span>
          </li>
        )}
        {details.description && (
          <li className="details-list-item">
            <span>Description</span>
            <span>{details.description}</span>
          </li>
        )}
        <li className="details-list-item">
          <span>Total photos</span>
          <span>{details.total_photos}</span>
        </li>
        <li className="details-list-item tag">
          <span>Tags</span>
          <div className="tags">
            {
              details.tags.map((tag) => (
                <div key={tag.id} >
                  <span>{tag.title},</span>
                </div>
              ))
            }
          </div>
        </li>
        <li className="details-list-item">
          <span>Photos in the collection</span>
          <span>
            Photos in
            <a
              href={details.links.html}
              target="_blank" rel="noreferrer"
            >
              " {details.title} "
            </a>
            collection
          </span>
        </li>
        <li className="details-list-item preview">
          <span>Preview Photos</span>
          <div className="image-container">
            {
              details.preview_photos.map((photo) => (
                  <img
                    key={photo.id}
                    src={photo.urls.regular}
                    alt={details.title}
                  />
              ))
            }
          </div>
        </li>
      </ul>
    </div>
  );
};

export default Details;
