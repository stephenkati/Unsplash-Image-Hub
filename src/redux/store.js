import { configureStore } from '@reduxjs/toolkit';
import DetailssSlice from './DetailsSlice';
import CollectionsReducer from './CollectionsSlice';

const store = configureStore({
  reducer: {
    collectionDetails: DetailssSlice,
    collections: CollectionsReducer,
  },
});

export default store;
