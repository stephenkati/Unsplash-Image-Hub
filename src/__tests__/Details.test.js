import React from 'react';
import { render, waitFor, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import fetchMock from 'jest-fetch-mock';
import store from '../redux/store';
import collectionsData from '../collectionsData';
import Details from '../components/Details';

describe('Collections Component', () => {

  beforeEach(() => {
    fetchMock.resetMocks();
  });

  it('should fetch collections and render them', async () => {
  
      // Mock the API request
      fetchMock.mockResponseOnce(JSON.stringify(collectionsData));
  
      const component = render(
        <Provider store={store}>
          <Details />
        </Provider>
      );
  
      await waitFor(() => screen.getByText(`Collection Stat's`));
      expect(component).toMatchSnapshot();
    });

  });
  