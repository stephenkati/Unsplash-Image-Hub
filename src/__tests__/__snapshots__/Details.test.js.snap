// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for Details snapshot for details file 1`] = `undefined`;
